### PR TITLE
build: improve roachtest build scripts

### PR DESCRIPTION
**Pull request by RaduBerinde**
We improve the roachtest scripts to support building finer-grained
components: the `roachtest_compile_component.sh` script builds a
single component (cockroach, cockroach-ea, workload, libgeos,
roachtest). For each component we define the bazel arguments and build
products, minimizing duplication of other code. There should be no
functional changes.

Epic: none
Release note: None

### CODE REVIEW

**Comment by RaduBerinde** [comment]
Sample build that just produces the bits as artifacts: https://teamcity.cockroachdb.com/viewLog.html?buildId=11963447&buildTypeId=Cockroach_Nightlies_RoachtestBuildBinariesAmd64&tab=buildLog&branch_Cockroach_Nightlies=111251

**Comment by jlinder** [comment]
A few drive by thoughts:

- You might consider is using the existing [CI cockroach builds](https://teamcity.cockroachdb.com/project/Cockroach_Ci_Builds?branch=%3Cdefault%3E&mode=builds#all-projects) for each architecture when you get there.
- When we implement remote execution for the nightly jobs, the build work will automatically be distributed and go faster. It seems like that obviate the need for at least some of these changes.
    - RBE status: The ability to enable this for building binaries for all nightly builds should happen in the next few months. We have EngFlow and are working on getting the first job running in it (nightly stress), GitHub error reporting and some other supporting things.

**Comment by RaduBerinde** [comment]
Thanks, I think holding off for remote execution makes sense. I still like the refactoring of the build code here.. I can remove the PREBUILT env var handling.

**Comment by rickystewart** [comment]
> When we implement remote execution for the nightly jobs, the build work will automatically be distributed and go faster. It seems like that obviate the need for at least some of these changes.

I wouldn't throw the baby out with the bathwater. It is still good to not build binaries you don't need.

**Comment by RaduBerinde** [comment]
Updated to remove the "prebuilt" support. This is now just a cleanup change with no functional changes to any build.

File: build/teamcity/cockroach/nightlies/roachtest_compile_component.sh
Comment (srosenberg) [inline build/teamcity/cockroach/nightlies/roachtest_compile_component.sh:56]: Nit: maybe `artifacts` since it's a bit more standard?

**Review by srosenberg — APPROVED** [review]
Nice!

**Comment by RaduBerinde** [comment]
Trying a build with low select probability before merging https://teamcity.cockroachdb.com/viewQueued.html?itemId=12046489

**Pull request merged**
